src/main/java/uk/gov/hmcts/reform/pip/data/management/service/CourtService.java

```diff
@@ -0,0 +1,94 @@
+package uk.gov.hmcts.reform.pip.data.management.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.pip.data.management.database.CourtsAndHearings;
+import uk.gov.hmcts.reform.pip.data.management.errorhandling.exceptions.CourtNotFoundException;
+import uk.gov.hmcts.reform.pip.data.management.models.Court;
+import uk.gov.hmcts.reform.pip.data.management.models.CourtMethods;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Service to handle the retrieval and filtering of courts.
+ */
+@Service
+@Slf4j
+public class CourtService {
+
+    @Autowired
+    private CourtsAndHearings courtsAndHearings;
+
+    @Autowired
+    private FilterService filterService;
+
+    /**
+     * Gets all courts.
+     *
+     * @return List of Courts
+     */
+    public List<Court> getAllCourts() {
+        return alphabetiseCourts(courtsAndHearings.getListCourts());
+    }
+
+    /**
+     * Handles request to search for a court by the court name.
+     *
+     * @param input the court name to search for
+     * @return Court of the found court
+     * @throws CourtNotFoundException when no court was found with the given search input
+     */
+    public Court handleSearchCourt(String input) {
+        List<Court> foundCourt = filterService.filterCourts(input, CourtMethods.NAME.methodName);
+        if (foundCourt.isEmpty()) {
+            throw new CourtNotFoundException(String.format("No court found with the search: %s", input));
+        } else {
+            return filterService.filterCourts(input, CourtMethods.NAME.methodName).get(0);
```

Comment:
Could we switch this to get the first court from the court list declared on line 49?

Saves us needing to call the filterCourts method again.

Reply:
good spot! meant to do that 😂